### PR TITLE
RACBlockTrampoline: remove `NSInvocation` usage.

### PR DESCRIPTION
**Pull request by StatusReport**
There are two optimizations that are done in this commit:
1. Avoid allocating a `RACBlockTrampoline` object on every invocation.
2. Do not use the very slow `NSInvocation`, and instead hard-code the
block calls.

### CODE REVIEW

File: ReactiveObjC/RACBlockTrampoline.m [hunk 1]
```diff
@@ -9,147 +9,44 @@
...
-		case 7: return @selector(performWith:::::::);
-		case 8: return @selector(performWith::::::::);
-		case 9: return @selector(performWith:::::::::);
-		case 10: return @selector(performWith::::::::::);
-		case 11: return @selector(performWith:::::::::::);
-		case 12: return @selector(performWith::::::::::::);
-		case 13: return @selector(performWith:::::::::::::);
-		case 14: return @selector(performWith::::::::::::::);
-		case 15: return @selector(performWith:::::::::::::::);
+id RACInvokeBlock(id block, RACTuple *args) {
```
Comment (alexger) [inline ReactiveObjC/RACBlockTrampoline.m]: static

Comment (alexger) [inline ReactiveObjC/RACBlockTrampoline.m]: I'd try static inline and move to .h file

**Comment by barakyoresh** [comment]
Up to Alex's concerns, LGTM.

File: ReactiveObjC/RACBlockTrampoline.m (hunk 1 above)
Comment (StatusReport) [inline ReactiveObjC/RACBlockTrampoline.m]: Done.

**Comment by StatusReport** [comment]
Done.

**Pull request merged**